### PR TITLE
federation v2 support

### PR DESCRIPTION
**Pull request by dariuszkuc**

### CODE REVIEW

**Comment by dariuszkuc** [comment]
```shell
# re-generate schema
dotnet run --project compatibility/Products.csproj schema export --output products.graphql

# run compatibility results
npx fedtest docker --compose compatibility/docker-compose.yaml --schema compatibility/products.graphql --debug
```

Results
```
*************
Federation v1 compatibility
*************
_service PASS
@key (single) PASS
@key (multi) PASS
@key (composite) WARNING
repeatable @key PASS
@requires PASS
@provides PASS
federated tracing WARNING

*************
Federation v2 compatibility
*************
@link PASS
@shareable PASS
@tag PASS
@override PASS
@inaccessible PASS
@composeDirective WARNING
@interfaceObject PASS
```

TLDR need to think to think about how to support `@composeDirective`.

File: src/Federation/One/ExternalDirectiveType.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 using ApolloGraphQL.Federation.HotChocolate.Constants;
 using ApolloGraphQL.Federation.HotChocolate.Properties;
 
-namespace ApolloGraphQL.Federation.HotChocolate;
+namespace ApolloGraphQL.Federation.HotChocolate.One;
```
Comment (dariuszkuc) [inline src/Federation/One/ExternalDirectiveType.cs]: I think we should change the namespacing to `ApolloGraphQL.HotChocolate.Federation` otherwise its very odd package naming when supporting v1/v2 schemas....

File: src/Federation/FederationTypeInterceptor.cs [hunk 2]
```diff
@@ -295,4 +300,128 @@ private void CompleteReferenceResolver(ObjectTypeDefinition typeDef)
         objectTypeDefinition.Directives.Add(
             new DirectiveDefinition(directiveNode));
     }
+
+    private void AddInaccessibleMarkers(DefinitionBase? definition)
```
Comment (dariuszkuc) [inline src/Federation/FederationTypeInterceptor.cs:238]: this is a pretty ugly workaround as currently `HotChocolate` does not allow you to create directive attributes that would be applicable to multiple targets, since `@tag`/`@inaccessible` are technically applicable on `FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ENUM | ENUM_VALUE | SCALAR | INPUT_OBJECT | INPUT_FIELD_DEFINITION | ARGUMENT_DEFINITION` we would need separate attribute per each possible location....
